cdap-app-fabric/src/main/java/co/cask/cdap/config/PreferencesService.java

```diff
@@ -154,12 +159,6 @@ private void setConfig(ProfileDataset profileDataset, PreferencesDataset prefere
     Optional<ProfileId> oldProfile = SystemArguments.getProfileIdFromArgs(namespaceId, oldProperties);
     preferencesDataset.setPreferences(entityId, propertyMap);
```

Comment:
earlier these steps happened before we publish like mentioned in the comment, is it safe to publish before setting preferences ?

Reply:
yeah, the message publish happens in a transaction so it's fine. The comment is confusing though I will change it

Reply:
actually, I think moving it up here made the flow more confusing, I'm going to move it back, even though it will result in an extra if condition.